### ISSUE TITLE
Why are pointers required for options?

### ISSUE THREAD

**Issue by gmenih341** [zmb3/spotify#60]
https://github.com/zmb3/spotify/blob/master/spotify.go#L235

I'm having trouble understanding why pointers are required for options? Especially for arguments like `Offset` and `Limit`.

**Comment by zmb3** [comment]
Because these options are all optional. You can specify offset and country but omit limit, for example.

**Comment by gmenih341** [comment]
But since Options are made out of primitive types, wouldn't it make more sense to check for zero values?

For example, in [this line](https://github.com/zmb3/spotify/blob/master/spotify.go#L258), couldn't we just check for zero value, instead of nil? I'm pretty new to Go, and I only skimmed through this package a bit, so bare with me.

If this make sense, I'm more than happy to do a PR.

**Comment by zmb3** [comment]
Perhaps, but there’s little point in breaking the API now.

**Comment by Strum355** [comment]
Personally, i think there should be a breaking change at some stage to change this. Having them all be pointers is a bit of a nuisance, and I've seen multiple people complain about it

**Comment by andrewpillar** [comment]
I agree with @Strum355. This breaking API change would be worth it. When working with primitive types you have more context around the variable you're working with when checking for its' existence. Sure it's nice, and easier to write

```go
if option.Limit == nil {

}
```

But then you lose the context around what `option.Limit` actually is. Furthermore I think this adds some needless verbosity to the API when it comes to creating a new Option struct.

Consider both of the following

```go
// With pointer values
limit := 5
country := "finland"

options := new(Spotify.Options)
options.Country = &country
options.Limit = &limit
```

```go
// Without pointer values
options := Spotify.Options{}
options.Country = "finland"
options.Limit = 5
```

I think this breaking change would be a long-term benefit to the API as a whole.

**Comment by zmb3** [comment]
No doubt that it’s cleaner without the pointers. There are lots of things I’d like to do for a v1 version of this library, for example I want to use the functional options pattern and make better use of context.Context. I simply don’t have the time to spend on it right now. 

Happy to help with PRs or even grant some of you folks contributor access if you want to help shape this.

**Comment by andrewpillar** [comment]
Opened a PR here #61.

**Comment by strideynet** [comment]
This will be resolved using functional options in 2.0
